source/module_esolver/esolver_ks_lcao_tddft.cpp

```diff
@@ -293,22 +294,17 @@ void ESolver_KS_LCAO_TDDFT::update_pot(const int istep, const int iter)
         }
     }
 
-    if (this->conv_elec)
+    if (elecstate::ElecStateLCAO<std::complex<double>>::out_wfc_lcao &&
+        (this->conv_elec || iter == GlobalV::SCF_NMAX) &&
+        (istep % GlobalV::out_interval == 0) )
     {
-        if (elecstate::ElecStateLCAO<std::complex<double>>::out_wfc_lcao)
-        {
-            elecstate::ElecStateLCAO<std::complex<double>>::out_wfc_flag
-                = elecstate::ElecStateLCAO<std::complex<double>>::out_wfc_lcao;
-        }
-        for (int ik = 0; ik < kv.nks; ik++)
-        {
-            if (istep % GlobalV::out_interval == 0)
-            {
-                this->psi[0].fix_k(ik);
-                this->pelec->print_psi(this->psi[0], istep);
-            }
-        }
-        elecstate::ElecStateLCAO<std::complex<double>>::out_wfc_flag = 0;
+            ModuleIO::write_wfc_lcao(elecstate::ElecStateLCAO<std::complex<double>>::out_wfc_lcao,
+                           this->psi[0],
+                           this->pelec->ekb,
+                           this->pelec->wg,
+                           this->pelec->klist->kvec_c,
+                           this->LOWF.ParaV,
```

Comment:
Could you avoid using LOWF.ParaV since this class will be removed soon?

Reply:
> It is hard to avoid using ParaV, because collect of LCAO psi needs some detail of 2d-block parallel. If ParaV is removed, how to store the information of 2d-block parallel?

You can use ParaV, but you can avoid using LOWF.ParaV

Reply:
Ok, I got it.